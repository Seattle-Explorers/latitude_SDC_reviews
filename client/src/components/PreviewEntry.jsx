import React from 'react';
import moment from 'moment';

// Styles import
import { TopStyle, AvatarStyle, TopText, Date } from './styles/generalUse.style';
import { PreviewWrapper } from './styles/staticStyles.style';

class PreviewEntry extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      readMore: null,
    }

    this.handleClick = this.handleClick.bind(this);
  }

  componentDidMount() {
    const { review } = this.props;
    const { body } = review;

    if (review.body.length > 180) {
      this.setState({
        readMore: true,
        display: body.slice(0,180)
      })
    } else {
      this.setState({
        readMore: false,
        display: body
      })
    }
  }

  handleClick() {
    const { review } = this.props;
    const { body } = review;

    this.setState({
      readMore: false,
      display: body
    })
  }

  render() {
    const { review } = this.props;
    const imageURL = `https://dteamdp.s3-us-west-2.amazonaws.com/pug${review.dp}.jpg`
    const date = moment(review.date).format('MMMM YYYY');

    if(this.state.readMore) {
      return(
        <PreviewWrapper>
          <TopStyle>
            <AvatarStyle src={imageURL} />
            <TopText>
              <b>{review.reviewer_name}</b>
              <br></br>
              <Date>{date}</Date>
            </TopText>
          </TopStyle>
          {`${this.state.display}...     `}   
          <a href='#' onClick={this.handleClick}><b>read more</b></a>
        </PreviewWrapper>
      )
    } else {
      return(
        <PreviewWrapper>
          <TopStyle>
            <AvatarStyle src={imageURL} />
            <TopText>
              <b>{review.reviewer_name}</b>
              <br></br>
              <Date>{date}</Date>
            </TopText>
          </TopStyle>
          {this.state.display}
        </PreviewWrapper>
      )
    }
  }
}

export default PreviewEntry;