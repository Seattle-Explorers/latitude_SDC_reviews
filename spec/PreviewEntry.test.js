import 'jsdom-global/register';
import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-enzyme';
import 'jest-styled-components';
import { mount } from 'enzyme';
import styled from 'styled-components';

import { reviews } from '../__mocks__/mockData.js';
import PreviewEntry from '../client/src/components/PreviewEntry.jsx';

describe('Preview Entry Components', () => {

  const AvatarStyle = styled.img`
  width: 50px;
  height: 50px;
  border-radius: 50%;
  flex-direction: row;
`
const entry = reviews[0];

  it('should render', () => {
    const component = <PreviewEntry review={entry} />;

    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('should render an image from S3', () => {
    
    const url = `https://dteamdp.s3-us-west-2.amazonaws.com/pug11.jpg`

    const wrapper = mount(<PreviewEntry review={entry} />);
    expect(wrapper.find('img').prop('src')).toEqual(url)

  })
})