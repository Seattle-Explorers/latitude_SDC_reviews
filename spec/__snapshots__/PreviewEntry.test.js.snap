// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preview Entry Components should render 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 10px;
}

.c2 {
  border-radius: 50%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 50px;
  margin-right: 10px;
  width: 50px;
}

.c3 {
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c4 {
  color: #B3B3B3;
}

.c0 {
  margin: 50px 15px 25px 50px;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <img
      className="c2"
      src="https://dteamdp.s3-us-west-2.amazonaws.com/pug11.jpg"
    />
    <div
      className="c3"
    >
      <b>
        Maureen
      </b>
      <br />
      <span
        className="c4"
      >
        September 2014
      </span>
    </div>
  </div>
  111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111...     
  <a
    href="#"
    onClick={[Function]}
  >
    <b>
      read more
    </b>
  </a>
</div>
`;
